tools/test_bazel.sh

```diff
@@ -0,0 +1,27 @@
+# This script tests the project with different versions of Bazel and compilers
+# locally. This avoids waiting on the CI to run the tests.
+
+# Version
+# -------
+# - Version >= 7 are supported
+# - Version <= 6 fail with the error:
```

Comment:
I wonder if passing `--enable_bzlmod` would fix this issue. iirc bzlmod wasn't defaulted to on until Bazel 7..

Reply:
It fixes the issue! Thanks!

Unfortunately, I still make it work on Bazel 6.0.0. I get:
```
external/googletest~1.14.0.bcr.1/googletest/include/gtest/internal/gtest-port.h:279:2: error: #error C++ versions less than C++14 are not supported.
```

Unsure why. Probably a problem from `googletest`.
```
ERROR: /home/arthursonzogni/.cache/bazel/_bazel_arthursonzogni/fd1242e4c381a003bc1e116d9c36c824/external/googletest~1.14.0.bcr.1/BUILD.bazel:80:11: Compiling googlemock/src/gmock-cardinalities.cc failed: (Exit 1): gcc failed: error executing command (from target @googletest~1.14.0.bcr.1//:gtest) /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer '-std=c++0x' -MD -MF ... (remaining 34 arguments skipped)
```

Reply:
I guess something's defaulting to C++14, either Bazel or an older compiler. You could try adding `build --cxxopt=-std=c++17`* to your `.bazelrc`, should fix it. Note that when people depend on FTXUI using Bazel, your `.bazelrc` will be ignored, and only the one in the top-level repository will be used.

*or this if you want things to be happy on Windows as well since they use `:` instead of `=` in their std flag:
```
build --enable_platform_specific_config
build:linux --cxxopt=-std=c++17
build:macos --cxxopt=-std=c++17
build:windows --cxxopt=-std:c++17
```

Reply:
Thanks! I will use the `.bazelrc` and abandon specifying the standard per target. This will be on dependants to do it.

Too bad Bazel doesn't support specifying the C++ standard for a target like CMake does.